TimescaleDB: Add psql to timescaledb migration scripts (#1364)
hashgraph/hedera-mirror-node: docs/operations.md

sudo journalctl -fu hedera-mirror-importer.service
```

### v1 to v2 Data Migration

To support time series logic the Mirror Node DB schema shifted from PostgeSQL (v1) to TimescaleDB (v2).
[Migrating from a Different PostgreSQL Database](https://docs.timescale.com/latest/getting-started/migrating-data#different-db) highlights the general recommended data migration steps when moving to TimescaleDB.

For mirror node operators running v1 db schema, the following steps can be taken to upgrade to v2.

> 1. Setup a new TimescaleDB database

Reviewer: ```suggestion
1. Set up a new TimescaleDB database
```

Author: Updated

    A new TimescaleDB server must be spun up.

    Refer to Mirror Node [DB Installation](installation.md#database-setup) for manual instructions.

    To use the Mirror Node configured docker container, simply run:

    ```shell script
    $ docker-compose up timescaledb
    ```

    Refer to [TimescaleDB Installation Instructions](https://docs.timescale.com/latest/getting-started/installation) for other installation options.

    > **_NOTE:_** The following steps assume the database, users and schema have been created as detailed above

2. Configure migration properties

    The configuration file `hedera-mirror-importer/src/main/resources/db/scripts/timescaledb/migration.config` contains db variables for easy running.
    These options include variables such as db names, passwords, users, hosts for both the existing db and the new db.

    Update these values appropriately for your db setup.

3. Run migration script

    From the `hedera-mirror-importer/src/main/resources/db` directory run the `migration.sh` script
    ```shell script
    $ ./scripts/timescaledb/migration.sh
    ```

>    The script uses successive `psql` connections to backup, configure and restore data on the new database nodes.

Reviewer: ```suggestion
   The script uses successive `psql` connections to back up, configure and restore data on the new database nodes.
```

Author: Updated

   First it copies over the `flyway_schema_history` table, to maintain migration history.

>    It then utilizes the migration sql script used by normal flyway operations to create the new tables and then creates the Timescale hyper tables based on these.

Reviewer: nit: I believe hypertable is always one word.
'```suggestion
   It then utilizes the migration sql script used by normal flyway operations to create the new tables and then creates the Timescale hypertables based on these.
```

Author: Fixed

sudo journalctl -fu hedera-mirror-importer.service
```

### v1 to v2 Data Migration

To support time series logic the Mirror Node DB schema shifted from PostgeSQL (v1) to TimescaleDB (v2).
[Migrating from a Different PostgreSQL Database](https://docs.timescale.com/latest/getting-started/migrating-data#different-db) highlights the general recommended data migration steps when moving to TimescaleDB.

For mirror node operators running v1 db schema, the following steps can be taken to upgrade to v2.

> 1. Setup a new database container using TimeScale

Author: There's a section in `Installation.md` for PostgreSQL. I can note that as V1 and add a V2 section for `TimescaleDB`.
Will refer to that from here, and with the removal of the override file this should be simpler

> 1. Setup a new database container using TimeScale

Author: Updated

    In the docker-compose case this file is already mounted under `/docker-entrypoint-initdb.d/` on the docker container and run on startup.

    This may be run manually against the db node if not using docker-compose:
    ```shell script
    $ psql "dbname=mirror_node host=localhost user=mirror_importer password=mirror_importer_pass port=6432" -f hedera-mirror-importer/src/main/resources/db/scripts/init_v2.sql
    ```

    > **_NOTE:_** The following steps assume the database, users and schema have been created as detailed above

3. Configure migration properties

    The configuration file `hedera-mirror-importer/src/main/resources/db/scripts/time-scale-migration/migration.config` contains db variables for easy running.
    These options include variables such as db names, passwords, users, hosts for both the existing db and the new db.

    Update these values appropriately for your db setup.

4. Run migration script

    From the `hedera-mirror-importer/src/main/resources/db` directory run the `timeScaleDbMigration.sh` script
    ```shell script

>     $ ./scripts/time-scale-migration/timeScaleDbMigration.sh

Reviewer 2: Seems a bit verbose and duplicates TimescaleDB. What about `scripts/timescaledb/migrate.sh`?

Author: Updated

    ```

   The script uses successive `psql` connections to backup, configure and restore data on the new database nodes.
   First it copies over the `flyway_schema_history` table, to maintain migration history.

>    It then utilizes the migration sql script used by normal flyway operations to create the new tables and then creates the Timescale hyper tables based on these.

Author: Will update to use `TimescaleDB` and `timescaledb` only

>    It then utilizes the migration sql script used by normal flyway operations to create the new tables and then creates the Timescale hyper tables based on these.

Author: Updated

- `/swagger/ui` - Metrics dashboard
- `/swagger/stats` - Aggregated statistics
- `/swagger/metrics` - Prometheus formatted metrics

> 

Author: Reverted
